File: algorithm/10844.py
#쉬운 계산 수
MOD = 1000000000

n = int(input())

cache = [[0]*10 for _ in range(101)]
for i in range(1,10):
  cache[1][i] = 1


for i in range(2,101):
  for j in range(10):
    if j > 0:
      cache[i][j] += cache[i-1][j-1]
      cache[i][j] %= MOD
    if j < 9:
      cache[i][j] += cache[i-1][j+1]
      cache[i][j] %= MOD




ans = 0
for j in range(10):
  ans += cache[n][j]
  ans %= MOD
print(ans)

File: algorithm/10844.js
const num = parseInt(require('fs')
      .readFileSync('/dev/stdin', 'utf-8')
      .trim())

let dp = [[0,1,1,1,1,1,1,1,1,1]]
for(let i = 0;i<num-1;i++){
  dp.push(new Array(9).fill(0))
}
let ans = 0
const mod = 1000000000
for(let i = 1;i<num;i++){
  for(let j = 0;j<10;j++){
    if(j == 0){
      dp[i][j] = (dp[i-1][j+1])%mod
    } else if(j == 9){
      dp[i][j] = (dp[i-1][j-1])%mod
    }else {
      dp[i][j] = (dp[i-1][j-1] + dp[i-1][j+1])%mod
    }

  }
}

for(let i = 0;i<10;i++){
  ans = (ans + dp[num-1][i])%mod
}
console.log(ans%mod)